js/data/bucket/circle_bucket.js

```diff
@@ -116,5 +116,18 @@ CircleBucket.prototype.addFeature = function(feature) {
         }
     }
 
-    this.populatePaintArrays('circle', globalProperties, feature.properties, startGroup, startIndex);
+    var groups = this.arrayGroups['circle'];
+    var range = {
+        startGroup: startGroup.index,
+        startVertex: startIndex,
+        endGroup: groups.length - 1,
+        endVertex: groups[groups.length - 1].layoutVertexArray.length - 1
+    };
```

Comment:
Do you think it would be simpler to combine the group index and the vertex index into a single number? 

 - group index is `Math.floor(index / ArrayGroup.MAX_VERTEX_ARRAY_LENGTH)`
 - vertex index is `index % ArrayGroup.MAX_VERTEX_ARRAY_LENGTH`